cpp/test/sg/fil_test.cu

```diff
@@ -742,33 +769,34 @@ class TreeliteFilTest : public BaseFilTest {
     } else {
       int left          = root + 2 * (node - root) + 1;
       int right         = root + 2 * (node - root) + 2;
-      float threshold   = dense_node.thresh();
       bool default_left = dense_node.def_left();
-      switch (ps.op) {
-        case tl::Operator::kLT: break;
-        case tl::Operator::kLE:
-          // adjust the threshold
-          threshold = std::nextafterf(threshold, -std::numeric_limits<float>::infinity());
-          break;
-        case tl::Operator::kGT:
-          // adjust the threshold; left and right still need to be swapped
-          threshold = std::nextafterf(threshold, -std::numeric_limits<float>::infinity());
-        case tl::Operator::kGE:
-          // swap left and right
-          std::swap(left, right);
-          default_left = !default_left;
-          break;
-        default: ASSERT(false, "comparison operator must be <, >, <= or >=");
+      float threshold;
+      if (dense_node.is_categorical()) {
+        uint8_t byte = 0;
+        for (int category = 0; category <= cat_sets_h.max_matching[dense_node.fid()]; ++category) {
+          if (category % BITS_PER_BYTE == 0) byte = cat_sets_h.bits[dense_node.set() + category / BITS_PER_BYTE];
```

Comment:
Could we put some braces around the then clauses in this and the next line as well?